`TestBotSSHMultiplexer` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/10062186397/job/27814072035?pr=44554

#### Relevant snippet


```
=== Failed
=== FAIL: lib/tbot TestBotSSHMultiplexer/server01.root:0\x00 (0.16s)
    tbot_test.go:1071: 
        	Error Trace:	/__w/teleport/teleport/lib/tbot/tbot_test.go:1071
        	Error:      	Received unexpected error:
        	            	ssh: handshake failed: EOF
        	Test:       	TestBotSSHMultiplexer/server01.root:0\x00
{"caller":"service/supervisor.go:262","component":"proc:2","level":"debug","message":"Service is completed and removed.","pid":"127223.2","service":"auth.server_info","timestamp":"2024-07-23T16:00:17Z"}

=== FAIL: lib/tbot TestBotSSHMultiplexer/server01.root:0|root\x00 (0.16s)
{"caller":"service/supervisor.go:262","component":"proc:2","level":"debug","message":"Service is completed and removed.","pid":"127223.2","service":"common.rotate","timestamp":"2024-07-23T16:00:17Z"}
    tbot_test.go:1071: 
        	Error Trace:	/__w/teleport/teleport/lib/tbot/tbot_test.go:1071
        	Error:      	Received unexpected error:
        	            	ssh: handshake failed: EOF
        	Test:       	TestBotSSHMultiplexer/server01.root:0|root\x00
```

Another hit: https://github.com/gravitational/teleport.e/actions/runs/17793949268/job/50577248456

```
    tbot_test.go:1168: 
        	Error Trace:	/__w/teleport.e/teleport.e/lib/tbot/tbot_test.go:1168
        	Error:      	Received unexpected error:
        	            	ssh: handshake failed: EOF
        	Test:       	TestBotSSHMultiplexer/server01.root:0\x00
--- FAIL: TestBotSSHMultiplexer/server01.root:0\x00 (0.06s)
```

@strideynet can someone take a look at this one?

--- a/lib/tbot/tbot_test.go
+++ b/lib/tbot/tbot_test.go
@@ -59,6 +59,7 @@ import (
 	"github.com/gravitational/teleport/lib/defaults"
 	"github.com/gravitational/teleport/lib/service"
 	"github.com/gravitational/teleport/lib/service/servicecfg"
+	"github.com/gravitational/teleport/lib/services/readonly"
 	"github.com/gravitational/teleport/lib/srv/db/common"
 	"github.com/gravitational/teleport/lib/srv/db/postgres"
 	apisshutils "github.com/gravitational/teleport/lib/sshutils"
@@ -1132,6 +1133,24 @@ func TestBotSSHMultiplexer(t *testing.T) {
 			_, err := os.Stat(filepath.Join(tmpDir, fileName))
 			require.NoError(t, err)
 		}
+	}, 10*time.Second, 100*time.Millisecond)
+
+	// We need to wait for the agent to be fully connected to the Proxy
+	// (e.g. visible in the reverse tunnel server) otherwise dials to it may
+	// result in "direct dialing to nodes not found in inventory is not
+	// supported".
+	require.EventuallyWithT(t, func(t *assert.CollectT) {
+		rts, err := process.GetReverseTunnelServer()
+		require.NoError(t, err)
+		cluster, err := rts.Cluster(ctx, "root")
+		require.NoError(t, err)
+		nw, err := cluster.NodeWatcher()
+		require.NoError(t, err)
+		got, err := nw.CurrentResourcesWithFilter(ctx, func(r readonly.Server) bool {
+			return r.GetHostname() == "server01"
+		})
+		require.NoError(t, err)
+		require.Len(t, got, 1)
 	}, 10*time.Second, 100*time.Millisecond)
 
 	targets := []string{

